pkg/kubelet/container/cache.go

```diff
@@ -93,12 +97,24 @@ func (c *cache) GetNewerThan(id types.UID, minTime time.Time) (*PodStatus, error
 	return d.status, d.err
 }
 
-// Set sets the PodStatus for the pod.
-func (c *cache) Set(id types.UID, status *PodStatus, err error, timestamp time.Time) {
+// Set sets the PodStatus for the pod only if the data is newer than the cache
+func (c *cache) Set(id types.UID, status *PodStatus, err error, timestamp time.Time) (updated bool) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	defer c.notify(id, timestamp)
+	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) {
```

Comment:
Nit: we can return early if no update is needed. 

```
...
        if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) {
		// Set the value in the cache only if it's not present already
		// or the timestamp in the cache is older than the current update timestamp
		if val, ok := c.pods[id]; ok && val.modified.After(timestamp) {
			return false
		}
	}
	c.pods[id] = &data{status: status, err: err, modified: timestamp}
	c.notify(id, timestamp)
	return true
}
```

Reply:
This looks better, thanks.

Reply:
Are we going to take this suggestion? I am still seeing the old code here.

Reply:
Oh sorry my bad, I missed that somehow. I will update the code.